sklearn/linear_model/tests/test_ridge.py

```diff
@@ -2012,6 +2012,96 @@ def test_ridge_positive_regression_test(solver, fit_intercept, alpha):
     assert np.all(model.coef_ >= 0)
 
 
+@pytest.mark.parametrize("positive", [[True, "foo"], [True, "True"], [0, True]])
+def test_ridge_positive_list_type_error_test(positive):
+    """
+    Test that Ridge raises ValueError when positive is List
+    and all values within positive list are not boolean.
+    """
+    X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+    coef = np.array([1, -10])
+    y = X.dot(coef)
+    model = Ridge(positive=positive, solver="lbfgs")
+
+    with pytest.raises(
+        ValueError,
+        match="'positive' must be either a boolean or a list of booleans",
+    ):
+        model.fit(X, y)
+
+
+@pytest.mark.parametrize("positive", [[True, True]])
+def test_ridge_positive_list_type_test(positive):
+    """
+    Test that Ridge raises ValueError when positive is List
+    and all values within positive list are boolean.
+    """
+    X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+    coef = np.array([1, -10])
+    y = X.dot(coef)
+    model = Ridge(positive=positive, solver="lbfgs")
+    model.fit(X, y)
```

Comment:
It seems that this test is redundant with `test_ridge_positive_list_length_test` and the docstring misleading.